src/coreclr/gc/gcconfig.h

```diff
@@ -75,8 +75,10 @@ class GCConfigStringHolder
     BOOL_CONFIG  (LogEnabled,             "GCLogEnabled",           NULL,                             false,             "Specifies if you want to turn on logging in GC")                                         \
     BOOL_CONFIG  (ConfigLogEnabled,       "GCConfigLogEnabled",     NULL,                             false,             "Specifies the name of the GC config log file")                                           \
     BOOL_CONFIG  (GCNumaAware,            "GCNumaAware",            NULL,                             true,              "Enables numa allocations in the GC")                                                     \
-    BOOL_CONFIG  (GCCpuGroup,             "GCCpuGroup",             "System.GC.CpuGroup",             false,             "Enables CPU groups in the GC")                                                           \
     BOOL_CONFIG  (GCLargePages,           "GCLargePages",           "System.GC.LargePages",           false,             "Enables using Large Pages in the GC")                                                    \
+    BOOL_CONFIG  (GCCpuGroup,             "GCCpuGroup",             "System.GC.CpuGroup",             false,             "Enables CPU groups in the GC")                                                           \
+    BOOL_CONFIG  (UseAllCpuGroups,        "Thread_UseAllCpuGroups", NULL,                             false,             "Enables using CPU group information for determining the processor count")                \
+    INT_CONFIG   (ProcessorCount,         "PROCESSOR_COUNT",        NULL,                             0,                 "Specifies the number of processors available for the process")                           \
     INT_CONFIG   (HeapVerifyLevel,        "HeapVerify",             NULL,                             HEAPVERIFY_NONE,   "When set verifies the integrity of the managed heap on entry and exit of each GC")       \
```

Comment:
can you please explain why these are in gcconfig.h? the thread one is not used by the GC and if GC wants a different # of heaps there's also another config (GCHeapCount).

Reply:
I mentioned that above.  At present two implementations of `GetCurrentProcessCpuCount` (one in GC and another in VM) disagree and may return different values.  The implementation in `utilcode\util.cpp` checks `CPUGroupInfo::CanEnableThreadUseAllCpuGroups`, while the implementation in `gc\windows\gcenv.windows.cpp` checks `CanEnableGCCPUGroups()` instead.  The former also requires `CLRConfig::EXTERNAL_Thread_UseAllCpuGroups` to be set.  My GC changes in this PR make the check in GC equivalent to the other implementation.  I think the two implementations should be kept in sync and return the same value.

Reply:
To answer the other part of the question, this new setting is introduced to mitigate a breaking change in some specific scenarios and allow quick experimentation.  Using existing GC settings is still the recommended and much preferred way to control GC behavior.  The new setting has to affect the `GetCurrentProcessCpuCount` implementation according to this note in #48094:
> This value is expected to affect `Environment.ProcessorCount` for managed code, and any analogues to it in the native runtime code.

Reply:
that seems unfortunate. @janvorli do you see a problem if we introduced a method on GCToEEInterface and get the value that way? so the GC side wouldn't need to know about configs it doesn't care about?

Reply:
I think that would work and it would be nicer. The reason for the two implementations was to enable independent enabling of CPU groups  in threadpool and in GC. But I don't see a reason why we would want to control that separately. @kouvel do you agree from the threadpool point of view?

Reply:
I think it would be reasonable. The "processor count" turned into an artifical number computed usign a complex policy, so it does not really make sense to have it on the GC/OS abstraction.

Reply:
> The reason for the two implementations was to enable independent enabling of CPU groups in threadpool and in GC. But I don't see a reason why we would want to control that separately.

From the threading side both must be set to use all CPU groups. I'm not sure but perhaps it's possible that there could be a benefit from enabling the GC to use all CPU groups but not for managed threads.

Reply:
I assume the consensus is to move `GetCurrentProcessCpuCount` to `GCToEEInterface`.  I will be happy to reduce code duplication.

Reply:
please do.